client/reader/recommended-sites/recommended-site.tsx

```diff
@@ -108,25 +101,102 @@ const RecommendedSite = ( {
 
 	useEffect( () => {
 		if ( railcar ) {
-			recordTracksRailcarRender( 'recommended_site', railcar, {
-				ui_algo: 'following_manage_recommended_site',
-				ui_position: uiPosition,
+			// reader: railcar, ui_algo: following_manage_recommended_site, ui_position, fetch_algo, fetch_position, rec_blog_id (incorrect: fetch_lang, action)
+			// submain: railcar, ui_algo: subscriptions_recommended_site, ui_position, fetch_algo, fetch_position, rec_blog_id
+			recordTrainTracksRender( {
+				railcarId,
+				uiAlgo: 'subscriptions_recommended_site',
+				uiPosition: uiPosition as unknown as number,
+				fetchAlgo: railcar?.fetch_algo as string,
+				fetchPosition: railcar?.fetch_position as number,
+				recBlogId: railcar?.ui_blog_id as string,
 			} );
 		}
-	}, [ railcar, uiPosition ] );
+	}, [ railcar, railcarId, uiPosition ] );
 
-	const recordEvent = ( eventName: ReaderEvent ) => {
-		const eventProps = {
-			blog_id: siteId,
-			feed_id: feedId,
-			source: READER_FOLLOWING_MANAGE_RECOMMENDATION,
-		};
+	const { isReaderPortal } = useSubscriptionManagerContext();
 
-		dispatch( recordReaderTracksEvent( eventName, eventProps ) as unknown as AnyAction );
+	const blog_id = siteId as unknown as string;
+	const feed_id = feedId as unknown as string;
```

Comment:
This is not going to ensure the actual type to be `string`. Those are only runtime checks. Instead you should:

```
const blog_id = String( siteId );
const feed_id = feedId !== undefined ? String( feedId ) : undefined;
```
However, it looks like `recordSiteIconClicked` requires the `feed_id` to be defined. This means that we should either return some default string value, when it is undefined, we should make the feed_id optional for  `recordSiteIconClicked` if this is really the case, or perhaps the `RecommendedSiteProps` should have the `feedId` as required.